### PR TITLE
Restore original external test branches

### PR DESCRIPTION
**Pull request by cameel**
~**This PR depends on #10429 and should be merged after it.**~ (done)
~**Also, please don't merge until branches in `solidity-external-tests` are updated**.~ (done)

This is just cleanup that could not be made a part of #10429. The base PR is running on branches suffixed with `_new` (e.g. `develop_070_new`) to avoid breaking tests on `develop`. This one is meant to switch back to the original branches after we update them to match the `_new` ones.

### CODE REVIEW

**Comment by chriseth** [comment]
What is the state here?

**Comment by cameel** [comment]
Now that I just #10429 is merged, I just need to push updated branches to external repos. Then tests will start passing.

**Comment by cameel** [comment]
Branches updated so this is now ready for review.

Summary of what I did in the external test repos:
- ENS, Colony and `util-contracts` (used by Gnosis v2) had upstream changes so I pulled them into the respective `master`/`develop` branches in our fork repos.
- In a few cases @hrkrshnn added new commits to his `update-0.8.0` branches since I last checked them. I pulled them into the `_080_new` branches so don't worry - they've been preserved.
- In repositories that had `_new` branches I moved the non `_new` variants to the same position and created a `legacy_` branch at their old location. So for example in Colony repo `legacy_develop_080` is now where `develop_080` used to be and `develop_080` matches `develop_080_new` now. We'll be able to delete `develop_080_new` once this PR is merged.
- The `update-0.8.0` branches were either removed or renamed to have `legacy-` prefix to make it clear that they are no longer relevant. We could just remove them since I cherry-picked everything important into the `_080` branches but I left some of them for now just in case I missed something.
- I created `solc-7_070` and `solc-7_080` in `util-contracts`. Both currently point at the same commit as the upstream `solc-7` so we could use it directly but we decided not to do that with `master` in ENS (see https://github.com/ethereum/solidity/pull/10453#discussion_r533320622) so I followed that convention here too.

**Review by ekpyron — APPROVED** [review]
Not sure I'm not missing something here, but seems to me like: fine to merge if tests pass.

**Comment by cameel** [comment]
`_070_new` branches removed from the repositories.

**Pull request merged**
